src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -171,7 +172,16 @@ struct TextAnalyzer final : IDWriteTextAnalysisSource, IDWriteTextAnalysisSink
 #pragma warning(suppress : 26455) // Default constructor may not throw. Declare it 'noexcept' (f.6).
 AtlasEngine::AtlasEngine()
 {
-    THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, _uuidof(_sr.d2dFactory), reinterpret_cast<void**>(_sr.d2dFactory.addressof())));
+#ifdef NDEBUG
+    THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, _sr.d2dFactory.addressof()));
+#else
+    {
+        D2D1_FACTORY_OPTIONS options;
+        options.debugLevel = D2D1_DEBUG_LEVEL_INFORMATION;
+        THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, options, _sr.d2dFactory.addressof()));
+    }
+#endif

```

Comment:
is it safe to just call the options version with an options of 0?

Reply:
It's not quite the same. The version without options passes a `nullptr` to the D2D factory constructor for options. I haven't looked any further into it, but I assume that this makes it possible to change the D2D debug level in the DirectX settings app, which wouldn't be possible otherwise. This code is also the "official" way that the docs suggest to use.